ocean_lib/models/test/test_side_staking.py

```diff
@@ -17,6 +17,19 @@
 )
 
 
+@pytest.mark.unit
+def test_properties(config, web3):
+    """Tests the events' properties."""
+    side_staking_address = get_address_of_type(config, "Staking")
+    side_staking = SideStaking(web3, side_staking_address)
+
+    assert side_staking.event_Vesting.abi["name"] == SideStaking.EVENT_VESTING
+    assert (
+        side_staking.event_VestingCreated.abi["name"]
+        == SideStaking.EVENT_VESTING_CREATED
+    )
+
+
 @pytest.mark.unit
 def test_side_staking(
```

Comment:
This method is now 362 lines long. This is _way_ too long. A good method - testing or implementation - should be max one page long (about 45 lines). Otherwise it's near impossible to inspect, review, or maintain.

This code needs to be refactored as such before I can even review it. 

Be sure to use conftest.py to re-use code and avoid DRY violations.

Reply:
It was treated different. I focused just on the vesting part which is [here](https://github.com/oceanprotocol/ocean.py/pull/758/files#diff-08860fbdc12cc1930e80f6143c04de0f728006f63cd4a0e12fdd7e97cf3c79b5). @md0x has a PR (#764 ) related to SS. In models we have unit tests for the contract functions themselves and we created a special folder for flows, where we added the functional tests and that's why I implemented the logic in `test_vesting.py`.